### PR TITLE
Fetch xpack binary only when xpack enabled

### PR DESCRIPTION
**Pull request by deric**
Currently Elasic fetches unnecessary xpack archive even when it's disabled.

### CODE REVIEW

**Comment by gabrielhartmann** [comment]
@deric:  This looks fine to me.  Could you resolve conflicts please?

**Comment by deric** [comment]
@gabrielhartmann Sure, I've rebased the commit. Sorry this took so long, the message got lost in mail.

**Comment by gabrielhartmann** [comment]
@deric No problem.  I'm kicking the `elastic` test once.  If it fails again I'll send you logs.

**Comment by deric** [comment]
@gabrielhartmann Good, some logs would be nice. Is it possible to run those tests locally without a test cluster?

**Comment by nickbp** [comment]
Things start failing in a test where xpack is enabled after the service is running. The tasks start crashing with not being able to find the x-pack.zip file. The mesos fetcher logs likewise don't list an "x-pack-5.6.5.zip", so it looks like the issue may be related to that.

Actually, from having compared this change to the current `marathon.json.mustache`, it looks like the setting should be `{{TASKCFG_ALL_XPACK_ENABLED}}` rather than just `{{XPACK_ENABLED}}`. The prior is the environment in the _scheduler_ (where svc.yml is rendered) while the latter is the environment in the _tasks_ (where task configs are rendered and `cmd` is run).

Log:

```
I0107 01:38:32.675071  5944 fetcher.cpp:551] Fetcher Info: {"cache_directory":"\/tmp\/mesos\/fetch\/nobody","items":[{"action":"BYPASS_CACHE","uri":{"extract":true,"value":"https:\/\/infinity-artifacts-ci.s3.amazonaws.com\/autodelete7d\/beta-elastic\/20180107-004301-UvW6HJNOJ8CvODNs\/bootstrap.zip"}},{"action":"BYPASS_CACHE","uri":{"extract":true,"value":"https:\/\/artifacts.elastic.co\/downloads\/elasticsearch\/elasticsearch-5.6.5.tar.gz"}},{"action":"BYPASS_CACHE","uri":{"extract":true,"value":"https:\/\/downloads.mesosphere.com\/java\/jre-8u152-linux-x64.tar.gz"}},{"action":"BYPASS_CACHE","uri":{"extract":true,"value":"https:\/\/infinity-artifacts-ci.s3.amazonaws.com\/autodelete7d\/beta-elastic\/20180107-004301-UvW6HJNOJ8CvODNs\/elastic-scheduler.zip"}},{"action":"BYPASS_CACHE","uri":{"extract":true,"value":"https:\/\/github.com\/mesosphere\/elasticsearch-statsd-plugin\/releases\/download\/5.6.5.0\/elasticsearch-statsd-5.6.5.0.zip"}},{"action":"BYPASS_CACHE","uri":{"extract":true,"value":"https:\/\/github.com\/elastic\/elasticsearch-support-diagnostics\/releases\/download\/6.2\/support-diagnostics-6.2-dist.zip"}},{"action":"BYPASS_CACHE","uri":{"extract":true,"value":"https:\/\/infinity-artifacts-ci.s3.amazonaws.com\/autodelete7d\/beta-elastic\/20180107-004301-UvW6HJNOJ8CvODNs\/executor.zip"}},{"action":"BYPASS_CACHE","uri":{"extract":false,"output_file":"config-templates\/elasticsearch","value":"http:\/\/api.testintegrationelastic.marathon.l4lb.thisdcos.directory\/v1\/artifacts\/template\/a8cb68e5-5d91-4246-a0d1-746fda5528be\/data\/node\/elasticsearch"}}],"sandbox_directory":"\/var\/lib\/mesos\/slave\/slaves\/3f0c988a-d693-4287-9c86-b37266e7715d-S0\/frameworks\/3f0c988a-d693-4287-9c86-b37266e7715d-0005\/executors\/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0\/runs\/e85ea363-6a5e-4c83-af25-9f128bab320b\/containers\/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc","user":"nobody"}
I0107 01:38:32.678912  5944 fetcher.cpp:450] Fetching URI 'https://infinity-artifacts-ci.s3.amazonaws.com/autodelete7d/beta-elastic/20180107-004301-UvW6HJNOJ8CvODNs/bootstrap.zip'
I0107 01:38:32.678928  5944 fetcher.cpp:291] Fetching directly into the sandbox directory
I0107 01:38:32.678948  5944 fetcher.cpp:225] Fetching URI 'https://infinity-artifacts-ci.s3.amazonaws.com/autodelete7d/beta-elastic/20180107-004301-UvW6HJNOJ8CvODNs/bootstrap.zip'
I0107 01:38:32.678963  5944 fetcher.cpp:176] Downloading resource from 'https://infinity-artifacts-ci.s3.amazonaws.com/autodelete7d/beta-elastic/20180107-004301-UvW6HJNOJ8CvODNs/bootstrap.zip' to '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/bootstrap.zip'
I0107 01:38:32.975183  5944 fetcher.cpp:134] Extracted '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/bootstrap.zip' into '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc'
I0107 01:38:32.975222  5944 fetcher.cpp:603] Fetched 'https://infinity-artifacts-ci.s3.amazonaws.com/autodelete7d/beta-elastic/20180107-004301-UvW6HJNOJ8CvODNs/bootstrap.zip' to '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/bootstrap.zip'
I0107 01:38:32.975231  5944 fetcher.cpp:450] Fetching URI 'https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-5.6.5.tar.gz'
I0107 01:38:32.975239  5944 fetcher.cpp:291] Fetching directly into the sandbox directory
I0107 01:38:32.975260  5944 fetcher.cpp:225] Fetching URI 'https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-5.6.5.tar.gz'
I0107 01:38:32.975271  5944 fetcher.cpp:176] Downloading resource from 'https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-5.6.5.tar.gz' to '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/elasticsearch-5.6.5.tar.gz'
I0107 01:38:53.847257  5944 fetcher.cpp:134] Extracted '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/elasticsearch-5.6.5.tar.gz' into '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc'
I0107 01:38:53.847301  5944 fetcher.cpp:603] Fetched 'https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-5.6.5.tar.gz' to '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/elasticsearch-5.6.5.tar.gz'
I0107 01:38:53.847309  5944 fetcher.cpp:450] Fetching URI 'https://downloads.mesosphere.com/java/jre-8u152-linux-x64.tar.gz'
I0107 01:38:53.847316  5944 fetcher.cpp:291] Fetching directly into the sandbox directory
I0107 01:38:53.847332  5944 fetcher.cpp:225] Fetching URI 'https://downloads.mesosphere.com/java/jre-8u152-linux-x64.tar.gz'
I0107 01:38:53.847342  5944 fetcher.cpp:176] Downloading resource from 'https://downloads.mesosphere.com/java/jre-8u152-linux-x64.tar.gz' to '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/jre-8u152-linux-x64.tar.gz'
I0107 01:38:58.569540  5944 fetcher.cpp:134] Extracted '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/jre-8u152-linux-x64.tar.gz' into '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc'
I0107 01:38:58.569608  5944 fetcher.cpp:603] Fetched 'https://downloads.mesosphere.com/java/jre-8u152-linux-x64.tar.gz' to '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/jre-8u152-linux-x64.tar.gz'
I0107 01:38:58.569623  5944 fetcher.cpp:450] Fetching URI 'https://infinity-artifacts-ci.s3.amazonaws.com/autodelete7d/beta-elastic/20180107-004301-UvW6HJNOJ8CvODNs/elastic-scheduler.zip'
I0107 01:38:58.569633  5944 fetcher.cpp:291] Fetching directly into the sandbox directory
I0107 01:38:58.569655  5944 fetcher.cpp:225] Fetching URI 'https://infinity-artifacts-ci.s3.amazonaws.com/autodelete7d/beta-elastic/20180107-004301-UvW6HJNOJ8CvODNs/elastic-scheduler.zip'
I0107 01:38:58.569672  5944 fetcher.cpp:176] Downloading resource from 'https://infinity-artifacts-ci.s3.amazonaws.com/autodelete7d/beta-elastic/20180107-004301-UvW6HJNOJ8CvODNs/elastic-scheduler.zip' to '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/elastic-scheduler.zip'
I0107 01:39:00.878975  5944 fetcher.cpp:134] Extracted '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/elastic-scheduler.zip' into '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc'
I0107 01:39:00.879019  5944 fetcher.cpp:603] Fetched 'https://infinity-artifacts-ci.s3.amazonaws.com/autodelete7d/beta-elastic/20180107-004301-UvW6HJNOJ8CvODNs/elastic-scheduler.zip' to '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/elastic-scheduler.zip'
I0107 01:39:00.879029  5944 fetcher.cpp:450] Fetching URI 'https://github.com/mesosphere/elasticsearch-statsd-plugin/releases/download/5.6.5.0/elasticsearch-statsd-5.6.5.0.zip'
I0107 01:39:00.879042  5944 fetcher.cpp:291] Fetching directly into the sandbox directory
I0107 01:39:00.879076  5944 fetcher.cpp:225] Fetching URI 'https://github.com/mesosphere/elasticsearch-statsd-plugin/releases/download/5.6.5.0/elasticsearch-statsd-5.6.5.0.zip'
I0107 01:39:00.879097  5944 fetcher.cpp:176] Downloading resource from 'https://github.com/mesosphere/elasticsearch-statsd-plugin/releases/download/5.6.5.0/elasticsearch-statsd-5.6.5.0.zip' to '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/elasticsearch-statsd-5.6.5.0.zip'
I0107 01:39:01.581231  5944 fetcher.cpp:134] Extracted '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/elasticsearch-statsd-5.6.5.0.zip' into '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc'
I0107 01:39:01.581285  5944 fetcher.cpp:603] Fetched 'https://github.com/mesosphere/elasticsearch-statsd-plugin/releases/download/5.6.5.0/elasticsearch-statsd-5.6.5.0.zip' to '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/elasticsearch-statsd-5.6.5.0.zip'
I0107 01:39:01.581295  5944 fetcher.cpp:450] Fetching URI 'https://github.com/elastic/elasticsearch-support-diagnostics/releases/download/6.2/support-diagnostics-6.2-dist.zip'
I0107 01:39:01.581301  5944 fetcher.cpp:291] Fetching directly into the sandbox directory
I0107 01:39:01.581317  5944 fetcher.cpp:225] Fetching URI 'https://github.com/elastic/elasticsearch-support-diagnostics/releases/download/6.2/support-diagnostics-6.2-dist.zip'
I0107 01:39:01.581328  5944 fetcher.cpp:176] Downloading resource from 'https://github.com/elastic/elasticsearch-support-diagnostics/releases/download/6.2/support-diagnostics-6.2-dist.zip' to '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/support-diagnostics-6.2-dist.zip'
I0107 01:39:08.604723  5944 fetcher.cpp:134] Extracted '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/support-diagnostics-6.2-dist.zip' into '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc'
I0107 01:39:08.604768  5944 fetcher.cpp:603] Fetched 'https://github.com/elastic/elasticsearch-support-diagnostics/releases/download/6.2/support-diagnostics-6.2-dist.zip' to '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/support-diagnostics-6.2-dist.zip'
I0107 01:39:08.604802  5944 fetcher.cpp:450] Fetching URI 'https://infinity-artifacts-ci.s3.amazonaws.com/autodelete7d/beta-elastic/20180107-004301-UvW6HJNOJ8CvODNs/executor.zip'
I0107 01:39:08.604812  5944 fetcher.cpp:291] Fetching directly into the sandbox directory
I0107 01:39:08.604828  5944 fetcher.cpp:225] Fetching URI 'https://infinity-artifacts-ci.s3.amazonaws.com/autodelete7d/beta-elastic/20180107-004301-UvW6HJNOJ8CvODNs/executor.zip'
I0107 01:39:08.604838  5944 fetcher.cpp:176] Downloading resource from 'https://infinity-artifacts-ci.s3.amazonaws.com/autodelete7d/beta-elastic/20180107-004301-UvW6HJNOJ8CvODNs/executor.zip' to '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/executor.zip'
I0107 01:39:09.206929  5944 fetcher.cpp:134] Extracted '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/executor.zip' into '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc'
I0107 01:39:09.206969  5944 fetcher.cpp:603] Fetched 'https://infinity-artifacts-ci.s3.amazonaws.com/autodelete7d/beta-elastic/20180107-004301-UvW6HJNOJ8CvODNs/executor.zip' to '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/executor.zip'
I0107 01:39:09.206979  5944 fetcher.cpp:450] Fetching URI 'http://api.testintegrationelastic.marathon.l4lb.thisdcos.directory/v1/artifacts/template/a8cb68e5-5d91-4246-a0d1-746fda5528be/data/node/elasticsearch'
I0107 01:39:09.206985  5944 fetcher.cpp:291] Fetching directly into the sandbox directory
I0107 01:39:09.207105  5944 fetcher.cpp:225] Fetching URI 'http://api.testintegrationelastic.marathon.l4lb.thisdcos.directory/v1/artifacts/template/a8cb68e5-5d91-4246-a0d1-746fda5528be/data/node/elasticsearch'
I0107 01:39:09.207121  5944 fetcher.cpp:176] Downloading resource from 'http://api.testintegrationelastic.marathon.l4lb.thisdcos.directory/v1/artifacts/template/a8cb68e5-5d91-4246-a0d1-746fda5528be/data/node/elasticsearch' to '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/config-templates/elasticsearch'
I0107 01:39:09.213745  5944 fetcher.cpp:603] Fetched 'http://api.testintegrationelastic.marathon.l4lb.thisdcos.directory/v1/artifacts/template/a8cb68e5-5d91-4246-a0d1-746fda5528be/data/node/elasticsearch' to '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/config-templates/elasticsearch'
I0107 01:39:09.213770  5944 fetcher.cpp:608] Successfully fetched all URIs into '/var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc'
Exception in thread "main" java.io.FileNotFoundException: /var/lib/mesos/slave/slaves/3f0c988a-d693-4287-9c86-b37266e7715d-S0/frameworks/3f0c988a-d693-4287-9c86-b37266e7715d-0005/executors/data__4dd1ad8e-9810-4530-9868-d2c83a67d9b0/runs/e85ea363-6a5e-4c83-af25-9f128bab320b/containers/2a0aefc3-fcf5-42e4-a254-b42e6fd3f0fc/x-pack-5.6.5.zip (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at sun.net.www.protocol.file.FileURLConnection.connect(FileURLConnection.java:90)
	at sun.net.www.protocol.file.FileURLConnection.getInputStream(FileURLConnection.java:188)
	at org.elasticsearch.plugins.InstallPluginCommand.downloadZip(InstallPluginCommand.java:328)
	at org.elasticsearch.plugins.InstallPluginCommand.download(InstallPluginCommand.java:248)
	at org.elasticsearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:216)
	at org.elasticsearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:202)
	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:70)
	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:134)
	at org.elasticsearch.cli.MultiCommand.execute(MultiCommand.java:69)
	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:134)
	at org.elasticsearch.cli.Command.main(Command.java:90)
	at org.elasticsearch.plugins.PluginCli.main(PluginCli.java:47)
```

**Comment by nickbp** [comment]
TLDR: Looks like the change will currently leave tasks without the xpack.zip even when xpack is enabled. Need to change the template parameter name to match the scheduler's environment rather than the per-task environment.

**Comment by deric** [comment]
@nickbp Thanks Nicholas, I wasn't sure which one I should be using.
